### PR TITLE
chore(DatePicker): updated controlled examples

### LINKED ISSUE #9136: Bug - Calendar - misaligned button when validation text appears
**Describe the problem**
If an invalid date is entered, the reset/toggle calendar button is misaligned. This happens on the Controlled example and the Controlling the date picker calendar state example.

**How do you reproduce the problem?**
Go to https://www.patternfly.org/v4/components/date-picker#controlled and enter an invalid date

**Expected behavior**
The primary button on the left should maintain alignment with the input.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by thatblindgeye**
**What**: Closes #9136 

**Additional issues**:

### CODE REVIEW

**Review by jenny-s51 — APPROVED** [review]
This looks great @thatblindgeye and the updates work as expected. While testing I seem to have found a bug with the demos page and I'm curious if you're able to reproduce this - the page seems to jump to the wrong location when the calendar icon is clicked, though the location seems pretty inconsistent

https://github.com/patternfly/patternfly-react/assets/32821331/4f95eaca-a90d-45c1-971a-1de7da9cab2e

**Comment by thatblindgeye** [comment]
@jenny-s51 I do get that bug as well. It looks like it's first getting positioned at the bottom of the page first:

image

Before getting positioned relative to the datepicker input:

image

**Comment by thatblindgeye** [comment]
@nicolethoen based on the PR when the logic was first added yes, but it looks like it's been broken in v4 as well.

**Comment by nicolethoen** [comment]
Should we either remove the example for fix it? Maybe as a follow up issue? Seems to be revealing that there is a preexisting bug in the DatePicker. WDYT @tlabaj ?

**Comment by tlabaj** [comment]
@thatblindgeye @nicolethoen  yes this issue existed in v4. @mcarrano had expresses that it would be nice to get this one fixed for the v5 release.

File: packages/react-core/src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -233,7 +231,12 @@ const DatePickerBase = (
           if (buttonRef.current && buttonRef.current.contains(event.target as Node)) {
             return false;
           }
-          setPopoverOpen(false);
+
+          if (popoverOpen) {
+            event.stopPropagation();
+            setPopoverOpen(false);
+            hideFunction();
+          }
```
Comment (thatblindgeye) [inline packages/react-core/src/components/DatePicker/DatePicker.tsx:239]: This prevents the CalendarMonth from opening again when pressing the "Toggle" button to close it, and also allows clicking outside the CalendarMonth to close it.

**Review by gitdallas — APPROVED** [review]

**Review by mcarrano — APPROVED** [review]
Looks good to me!

**Pull request merged**